src/test/spec/retryable_writes/mod.rs

```diff
@@ -45,19 +46,20 @@ async fn run_unified() {
     run_spec_test_with_path(&["retryable-writes", "unified"], run_unified_format_test).await;
 }
 
-#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "tokio-runtime", tokio::test(flavor = "multi_thread"))]
 #[cfg_attr(feature = "async-std-runtime", async_std::test)]
 async fn run_legacy() {
     async fn run_test(test_file: TestFile) {
         for mut test_case in test_file.tests {
             if test_case.operation.name == "bulkWrite" {
                 continue;
             }
-            let hosts = &CLIENT_OPTIONS.get().await.hosts;
-            let options = test_case.client_options.map(|mut opts| {
-                opts.hosts = hosts.clone();
-                opts
-            });
+            let mut options = test_case.client_options.unwrap_or_default();
+            options.hosts = CLIENT_OPTIONS.get().await.hosts.clone();
+            if options.heartbeat_freq.is_none() {
+                options.heartbeat_freq = Some(MIN_HEARTBEAT_FREQUENCY);
```

Comment:
So this is actually the same problem that made us disable that other test: only the operation receives the error via failpoint. As far as the monitor that is streaming is concerned, nothing has happened, and so it continues to wait for the server to push a new response (heartbeatFrequencyMS / maxAwaitTimeMS) all the while ignoring check requests. Setting this really low tells the server to push updates more often so that the monitor can more quickly rediscover the server. In actual scenarios, if an operation is seeing a state change error, the monitor will likely have already seen it.